frontend/packages/data-portal/app/components/TablePageLayout.tsx

```diff
@@ -4,54 +4,121 @@ import { ReactNode, useEffect, useMemo } from 'react'
+  header?: ReactNode
+
+  tabs: TableLayoutTab[] // If there is only 1 tab, the tab selector will not show.
+  tabsTitle?: string
+
+  downloadModal?: ReactNode
+  drawers?: ReactNode
+}
+
+export interface TableLayoutTab {
+  title: string
+
+  filterPanel?: ReactNode
+
+  table: ReactNode
+  noResults?: ReactNode
+  pageQueryParam?: string
+
+  filteredCount: number
+  totalCount: number
+  countLabel: string // e.g. "objects" in "1 of 3 objects".
+}
+
+/** Standard page structure for browsing + filtering list(s) of objects. */
 export function TablePageLayout({
+  header,
+  tabs,
+  tabsTitle,
   downloadModal,
   drawers,
+}: TablePageLayoutProps) {
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const activeTabTitle = searchParams.get(QueryParams.TableTab)
+  const activeTab = tabs.find((tab) => tab.title === activeTabTitle) ?? tabs[0]
+
+  return (
+    <>
+      {downloadModal}
+
+      <div className="flex flex-col flex-auto">
+        {header}
+
+        {tabs.length > 1 && (
+          <>
+            {tabsTitle && <div className="text-sds-header-l">{tabsTitle}</div>}
+            <Tabs
+              value={activeTab.title}
+              onChange={(tabTitle: string) => {
+                setSearchParams((prev) => {
+                  prev.set(QueryParams.TableTab, tabTitle)
+                  return prev
+                })
+              }}
+              tabs={tabs.map((tab) => ({
+                label: (
+                  <>
+                    <span>{tab.title}</span>
+                    <span className="text-sds-gray-500 ml-[24px]">
+                      {tab.filteredCount}
+                    </span>
+                  </>
+                ),
+                value: tab.title,
+              }))}
+            />
+          </>
+        )}
+        <TablePageTabContent {...activeTab} />
+
+        {drawers}
+      </div>
+    </>
+  )
+}
+
+/** Table + filters for 1 tab. */
+function TablePageTabContent({
+  title,
+  filterPanel,
   filteredCount,
-  filters: filterPanel,
-  header,
-  noResults,
   table,
-  title,
+  noResults,
+  pageQueryParam = QueryParams.Page,
   totalCount,
-  type,
-}: {
-  downloadModal?: ReactNode
-  drawers?: ReactNode
-  filteredCount: number
-  filters?: ReactNode
-  header?: ReactNode
-  noResults?: ReactNode
-  table: ReactNode
-  title: string
-  totalCount: number
-  type: string
-}) {
+  countLabel,
+}: TableLayoutTab) {
   const [searchParams, setSearchParams] = useSearchParams()
-  const page = +(searchParams.get('page') ?? '1')
+  const page = +(searchParams.get(pageQueryParam) ?? '1')
 
   useEffect(() => {
     if (Math.ceil(filteredCount / MAX_PER_PAGE) < page) {
       setSearchParams(
         (prev) => {
-          prev.delete('page')
+          prev.delete(pageQueryParam)
           return prev
         },
         { replace: true },
       )
     }
-  }, [filteredCount, page, setSearchParams])
+  }, [filteredCount, page, pageQueryParam, setSearchParams])
```

Comment:
It's confusing that `page` and `pageQueryParam` are two different things.  Can we rename one?  
`tableTypeQueryParam`? `dataTypeQueryParam`?  🤔

Reply:
Renamed both.